main.go

```diff
@@ -114,6 +117,11 @@ func main() {
 		setupLog.Error(err, "failed to parse env params")
 		os.Exit(1)
 	}
+
+	if _, exists := os.LookupEnv("DEPLOY_NETWORK_POLICIES"); !exists {
```

Comment:
keeping the discussion on stale in this comment, and commenting only the logic:

here we are lying, because we say envParams has this value set, while it's really "env value plus some extra logic". If we go with this, I'd say it has to belong to where we make the call to create the network policies, not here where we are collecting the parameters.